br/pkg/restore/utils/merge.go

```diff
@@ -48,6 +51,14 @@ func MergeAndRewriteFileRanges(
 	for _, file := range files {
 		filesMap[string(file.StartKey)] = append(filesMap[string(file.StartKey)], file)
 
+		// Assert that it has the same end key.
+		if !bytes.Equal(filesMap[string(file.StartKey)][0].EndKey, file.EndKey) {
```

Comment:
why need this check? for some scenarios like compacted ssts may not satisfied.

Reply:
To assert range's start-key and end-key is the same as that of any file of the range.